lldb/source/Plugins/ExpressionParser/Swift/SwiftPersistentExpressionState.h

```diff
@@ -114,8 +119,15 @@ class SwiftPersistentExpressionState : public PersistentExpressionState {
 
   // This just adds this module to the list of hand-loaded modules, it doesn't
   // actually load it.
-  void AddHandLoadedModule(ConstString module_name) {
-    m_hand_loaded_modules.insert(module_name);
+  void AddHandLoadedModule(
+      ConstString module_name,
+      swift::AttributedImport<swift::ImportedModule> attributed_import) {
+    // TODO: Refactor to m_hand_loaded_modules.insert_or_assign when C++17 is
+    // available.
+    if (m_hand_loaded_modules.find(module_name) != m_hand_loaded_modules.end())
+      m_hand_loaded_modules.erase(module_name);
```

Comment:
Can you add a comment here why we erase the old entry? Also, do we need this function at all? If m_hand_loaded_modules were a DenseMap, wouldn't it be the same to write `m_hand_loaded_modules[module_name] = ...`?

Reply:
Okay we may need the function because it's called from outside the class, but the implementation could be a lot shorter :-)

Reply:
I switched to using a StringMap instead.